front/tests/chat/e2e_eval.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * Run an eval of our chat app with a test set of questions See:
+ * https://www.notion.so/dust-tt/Design-Doc-Chat-Eval-0b849d4c66564f1e9c0c31d537c96f78
+ *
+ *
+ * By default, runs on all questions. Use `--n NUMBER_OF_QUESTIONS` to change that (e.g. smaller evals)
+ * Use `--verbose` to print the successes as well as the failures
+ * Usage: from front directory
+ * `./admin/eval.sh [--verbose] [--n NUMBER_OF_QUESTIONS]`
+ *
+ *
+ */
+
+import * as fs from "fs";
+import * as path from "path";
+
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions/registry";
+import { runAction } from "@app/lib/actions/server";
+import { newChat } from "@app/lib/api/chat";
+import { Authenticator } from "@app/lib/auth";
+import { Ok } from "@app/lib/result";
+import { RunType } from "@app/types/run";
+
+type ChatEvalInputType = {
+  question: string;
+  rules: string[];
+  answer: string;
+};
+
+type EvalResultType = {
+  rule: string;
+  rule_respected: string;
+  explanation?: string;
+};
+
+async function chatEval(
+  numberOfQuestions: number | undefined,
+  verbose = false
+) {
+  /* Auth */
+  const workspacesId = process.env.LOCALHOST_WORKSPACE_ID as string;
+  const auth = await Authenticator.internalBuilderForWorkspace(workspacesId);
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Invalid workspace");
+  }
+
+  async function computeChatAnswer(input: ChatEvalInputType) {
+    /* Call the chat lib */
+    const chatRes = await newChat(
+      auth,
+      {
+        userMessage: input.question,
+        dataSources: null,
+        filter: null,
+        timeZone: "Europe/Paris",
+      },
+      false
+    );
+    let answer = undefined;
+    for await (const event of chatRes) {
+      if (
+        event.type === "chat_message_create" &&
+        event.message.role === "assistant"
+      ) {
+        answer = event.message.message;
+      } else if (event.type === "chat_session_create") {
+        if (verbose) console.log(event.session);
+      }
+    }
+
+    return answer ? answer : "There was an error. Please check with the team";
+  }
+
+  /* Load the JSON test data from data/chat-eval-inputs.jsonl */
+  let evalData = JSON.parse(
+    fs.readFileSync(path.join(__dirname, "data/chat-eval-inputs.json"), "utf8")
+  );
+  if (numberOfQuestions)
+    evalData = evalData.slice(0, numberOfQuestions) as ChatEvalInputType[];
```

Comment:
Reading that I understand this param is there to limit the number of questions we want to process in the json file. What is the use case for this? Why would I want only the top 3 questions to be asked?